sentry_sdk/profiler.py

```diff
@@ -729,46 +800,3 @@ def run(self):
             # after sleeping, make sure to take the current
             # timestamp so we can use it next iteration
             last = time.perf_counter()
-
-
-def _should_profile(transaction, hub):
-    # type: (sentry_sdk.tracing.Transaction, sentry_sdk.Hub) -> bool
-
-    # The corresponding transaction was not sampled,
-    # so don't generate a profile for it.
-    if not transaction.sampled:
-        return False
-
-    # The profiler hasn't been properly initialized.
-    if _scheduler is None:
-        return False
-
-    client = hub.client
-
-    # The client is None, so we can't get the sample rate.
-    if client is None:
-        return False
-
-    options = client.options
-    profiles_sample_rate = options["_experiments"].get("profiles_sample_rate")
-
-    # The profiles_sample_rate option was not set, so profiling
-    # was never enabled.
-    if profiles_sample_rate is None:
-        return False
-
-    return random.random() < float(profiles_sample_rate)
-
-
-@contextmanager
-def start_profiling(transaction, hub=None):
-    # type: (sentry_sdk.tracing.Transaction, Optional[sentry_sdk.Hub]) -> Generator[None, None, None]
-    hub = hub or sentry_sdk.Hub.current
-
-    # if profiling was not enabled, this should be a noop
-    if _should_profile(transaction, hub):
-        assert _scheduler is not None
-        with Profile(_scheduler, transaction, hub):
-            yield
-    else:
-        yield
```

Comment:
`hub.start_transaction(...)` will start the profile automatically if needed so this context manager is no longer necessary.